lib/template-loader.php

```diff
@@ -107,6 +107,42 @@ function gutenberg_find_template( $template_file ) {
 	if ( $template_query->have_posts() ) {
 		$template_posts            = $template_query->get_posts();
 		$_wp_current_template_post = array_shift( $template_posts );
+
+		// Build map of template slugs to their priority in the current hierarchy.
+		$slug_priorities = new stdClass();
+		foreach ( $slugs as $index => $slug ) {
+			$slug_priorities->{$slug} = $index;
+		}
+
+		// See if there is a theme block template with higher priority than the resolved template post.
+		$higher_priority_block_template_path     = null;
+		$higher_priority_block_template_priority = PHP_INT_MAX;
+		foreach ( glob( get_template_directory() . '/block-templates/*.html', 1 ) as $path ) {
+			$theme_block_template_priority = $slug_priorities->{basename( $path, '.html' )};
```

Comment:
```suggestion
			$theme_block_template_priority = $slug_priorities[ basename( $path, '.html' ) ];
```

Should this not also check if `isset( $slug_priorities[ basename( $path, '.html' ) ] )`?

Reply:
That's my thinking too, we need to wrap this in an `isset`.

Reply:
Yes, good catch!